spec/forms/dates_form_spec.rb

```diff
@@ -117,4 +117,45 @@
       end
     end
   end
+
+  describe "ensure departure date is no more than 14 days after landing date" do
+    let(:landing_date) { Date.today + 1.day }
+
+    context "when departure date is less than 14 days after landing date" do
+      let(:departure_date) { landing_date + 5.day }
+
+      let(:form) { DatesForm.new(landing_date: landing_date, departure_date: departure_date) }
+
+      it "should NOT flag an error" do
+        form.valid?
+
+        expect(form.errors).to_not include(:departure_date)
+      end
+    end
+
+    context "when departure date is 14 days after landing date" do
+      let(:departure_date) { landing_date + 14.day }
+
+      let(:form) { DatesForm.new(landing_date: landing_date, departure_date: departure_date) }
+
+      it "should NOT flag an error" do
+        form.valid?
+
+        expect(form.errors).to_not include(:departure_date)
+      end
+    end
+
+    context "when departure date is more than 14 days after landing date" do
+      let(:departure_date) { landing_date + 15.day }
+
+      let(:form) { DatesForm.new(landing_date: landing_date, departure_date: departure_date) }
+
+      it "should flag an error" do
+        form.valid?
+
+        expect(form.errors).to include(:departure_date)
+        expect(form.errors.full_messages.join).to match("Departure date must be within 14 days of landing date")
+      end
```

Comment:
you _might_ then think about reworking the description of the behaviour in terms of the "maximum permitted stay" (available as `DatesForm::MAXIMUM_PERMITTED_STAY_IN_DAYS`) e.g.

```
describe "ensure departure date is no more than the maximum permitted after landing date
  - context "when departure date is less than the maximum permitted after landing date"
  - context "when departure date is exactly the maximum permitted after landing date" 
  - context "when departure date is more than the maximum permitted after landing date"
```

However, I would say that your currently use of the literal value of 14 is clearer and more helpful in this case -- and that using `DatesForm::MAXIMUM_PERMITTED_STAY_IN_DAYS` in the spec would only add indirection at present...